enterprise/frontend/src/metabase-enterprise/tools/containers/ErrorDetail.jsx

```diff
@@ -0,0 +1,185 @@
+  const linkClass = urlVal === "" ? "" : "text-brand";
+  return [urlVal, linkClass];
+}
+
+function ErrorDetailDisplay(props) {
+  const { result } = props;
+  const resRow = getIn(result, ["data", "rows", 0]);
+  const resCols = getIn(result, ["data", "cols"]);
+  if (resRow && resCols) {
+    const nameToResCol = resCols.reduce(
+      (obj, x, idx) => Object.assign(obj, { [x.name]: idx }),
+      {},
+    );
+
+    const linkColumns = [
+      null,
+      "collection_id",
+      "database_id",
+      null,
+      "table_id",
+      null,
+      "user_id",
+      null,
+    ];
+
+    const ordinaryRows = [
+      "last_run_at",
+      "collection_name",
+      "database_name",
+      "schema_name",
+      "table_name",
+      "total_runs",
+      "user_name",
+      "updated_at",
+    ].map((x, idx) => {
+      const [urlVal, linkClass] = idxToUrl(
+        resRow,
+        resCols,
+        nameToResCol,
+        linkColumns[idx],
+      );
+      const formattedVal = formatValue(resRow[nameToResCol[x]], {
+        column: resCols[nameToResCol[x]],
+        jsx: true,
+        rich: true,
+        type: "cell",
+        local: true,
+      });
+      return (
+        <tr key={x}>
+          <td align="right" className="m0 mt1 text-medium">
+            {formatColumn(resCols[nameToResCol[x]])}
+          </td>
+          <td>
+            {
+              <Link to={urlVal} className={linkClass}>
+                {formattedVal}
+              </Link>
+            }
+          </td>
+        </tr>
+      );
+    });
+
+    const dashIdRows = resRow[nameToResCol.dash_name_str]
+      ?.split("|")
+      ?.map((x, idx) => (
+        <tr key={x}>
+          <td align="right" className="m0 mt1 text-medium">
+            {idx === 0 && formatColumn(resCols[nameToResCol.dash_name_str])}
+          </td>
+          <td className="text-bold">
+            {formatValue(x, { column: resCols[nameToResCol.dash_name_str] })}
+          </td>
+        </tr>
+      ));
+
+    const [cardUrlVal, cardLinkClass] = idxToUrl(
+      resRow,
+      resCols,
+      nameToResCol,
+      "card_id",
+    );
+
+    return [
+      <h2 className="PageTitle p1" key="card_name">
+        {
+          <Link to={cardUrlVal} className={cardLinkClass}>
+            {resRow[nameToResCol.card_name]}
+          </Link>
+        }
+      </h2>,
+      <div key="error_str" className="p1 text-code">
+        {resRow[nameToResCol.error_str]}
+      </div>,
+      <table key="table" className="ContentTable">
+        <tbody>{[ordinaryRows, dashIdRows]}</tbody>
+      </table>,
+    ];
+  } else {
+    return null;
+  }
+}
+
+export default function ErrorDetail(props) {
+  const { params } = props;
+  const cardId = parseInt(params.cardId);
+
+  const errorRetry = async cardId => {
+    await CardApi.query({ cardId: cardId });
+    // we're imagining that we successfully reran, in which case we want to go back to overall table
+    window.location = "/admin/tools/errors/";
```

Comment:
It would be better to connect the component to the store and dispatch `push` action so that the page is not fully reloaded